includes/components/class-admin.php

```diff
@@ -1681,6 +1817,51 @@ public function render_notices() {
 		// Add default notices.
 		$notices = [];
 
+        // Get unlicensed products.
+        $unlicensed_products = $this->get_unlicensed_products();
+
+        if ( $unlicensed_products['themes'] ) {
+            $notices['unlicensed_themes'] = [
+                'type'        => 'warning',
+                'dismissible' => true,
+                'text'        => sprintf(
+                /* translators: 1: themes names, 2: URL. */
+                    hp\sanitize_html( __( 'Functionality of %1$s themes has been disabled since you did not activate license key for them. You can set license key in the <a href="%2$s">Settings</a>.', 'hivepress' ) ),
+                    implode(
+                        ', ',
+                        array_map(
+                            function ( $theme ) {
+                                return $theme['name'];
+                            },
+                            $unlicensed_products['themes']
+                        )
+                    ),
+                    esc_url( admin_url( 'admin.php?page=hp_settings&tab=integrations' ) )
+                ),
+            ];
+        }
+
+        if ( $unlicensed_products['extensions'] ) {
+            $notices['unlicensed_extensions'] = [
+                'type'        => 'warning',
+                'dismissible' => true,
+                'text'        => sprintf(
+                /* translators: 1: extensions names, 2: URL. */
+                    hp\sanitize_html( __( 'Functionality of %1$s extensions has been disabled since you did not activate license key for them. You can set license key in the <a href="%2$s">Settings</a>.', 'hivepress' ) ),
+                    implode(
+                        ', ',
+                        array_map(
+                            function ( $extension ) {
+                                return hivepress()->get_name() . ' ' . $extension['name'];
+                            },
+                            $unlicensed_products['extensions']
+                        )
+                    ),
+                    esc_url( admin_url( 'admin.php?page=hp_settings&tab=integrations' ) )
+                ),
+            ];
+        }
+
```

Comment:
The overall workflow should be a bit different because currently it also blames customers who just purchased and installed the theme/extensions. For example someone purchases the theme and installs it, along with packaged extensions. Automatically disabling extensions is not ok because they haven't entered the license key yet, only the installation is complete. So the workflow is like this:

1. The customer installs and activates the theme with its extensions
2. He sees the non-dismissible admin notices to add the license key with a link
3. While the license key is not added, the extensions should be active but disabled (for now you can remove the extension paths from hivepress/v1/extensions, if it's empty in the constructor of the component then wrap it with another callback, e.g. https://developer.wordpress.org/reference/hooks/plugins_loaded/ Then the filter will contain all the paths for sure. This is not reliable but I'll figure out how to secure this later. Also, let me know if you have ideas to secure this, because another issue is that pirates can simply create a snippet to override the cached list of purchased theme/extensions.

Also, the main purpose of the theme is design, I thought that we can filter the wp_body_open hook and add a white DIV which overlays the whole website, or maybe add a notice at the bottom of the website (also via the "position:fixed") that the product is not licensed and a suggestion to add the license key.